src/transform/src/join_implementation.rs

```diff
@@ -557,15 +569,46 @@ mod delta_queries {
             // A viable delta query requires that, for every order,
             // there is an arrangement for every input except for
             // the starting one.
-            if !orders
+            let arrangement_counts = orders
                 .iter()
-                .all(|o| o.iter().skip(1).all(|(c, _, _)| c.arranged))
-            {
-                return Err(TransformError::Internal(String::from(
-                    "delta plan not viable",
-                )));
+                .map(|o| o.iter().skip(1).filter(|(c, _, _)| c.arranged).count())
+                .collect::<Vec<_>>();
+            let expected_number_of_arrangements = inputs.len() - 1;
```

Comment:
This is the expected number for a differential join, right? Could you please name it in a way that mention "differential"?